### ISSUE TITLE
Feature request: Override settings and rules for certain globs or file names

### ISSUE THREAD

**Issue by ngerritsen** [eslint/eslint#5609]
I am currently using eslint 2.2.0 with babel-eslint, and I want to have some different rules for my tests.
For instance, tests have the 'mocha' environment and in tests I allow deep nesting of callbacks (because of nested test suites). However my tests are not in one seperate folder but structured like this:

```
- app/
    - __tests__
    - components
        - __tests__
        - shared
            - __tests__
```

This structure works really good for me. However, this means I either would have to put an .eslint file in each single \_\_tests\_\_ folder. Or use a separate config like I currently do.

Currently I do the following:

```json
{
  "lint": "npm run lint:base && npm run lint:tests",
  "lint:base": "eslint .",
  "lint:tests": "eslint . -c .eslintrc.tests.js --ignore-path .eslintignore.tests"
}
```

This means that lint:base lints all files but ignores the ones in \_\_tests\_\_. lint:test lints ALL files with the less strict test rules (but this relies on test rules actually being less strict for all rules, which is not a given). Also this means my IDE will give errors in the test because it can only use one .eslintrc file at a time.

Anyone have any solution for this? Otherwise it would be nice to be able to specify rules for certain globs like:

```js
{
  rules: {
    'rule-a': 2,
    'rule-b': 2,
  }
  tests: { // (Would inherit from base)
    pattern: '__tests__/*.test.js',
    env: {
      mocha: true
    }
    rules: {
      'rule-a': 0
    }
  }
}
```

This is just a first idea of doing it. Any suggestions are welcome.

**Comment by platinumazure** [comment]
I think this would/could be covered by glob-based configuration, which is under discussion in issue #3611 right now.
